src/Description/SonataEnhancer.php

```diff
@@ -118,6 +120,27 @@ public function supports(PuliResource $resource)
         // sonata has dependency on ClassUtils so this is fine.
         $class = ClassUtils::getClass($payload);
 
-        return $this->pool->hasAdminByClass($class);
+        return null !== $this->getAdminByClass($class);
+
+    }
+
+    private function getAdminByClass($class)
+    {
+        if (isset($this->adminMap[$class])) {
+            return $this->adminMap[$class];
+        }
+
+        $_class = $class;
+        while ($_class && !$this->pool->hasAdminByClass($_class)) {
+            $_class = get_parent_class($_class);
+        }
+
+        if (false === $class) {
+            return;
```

Comment:
does that mean we will redo walking the class hierarchy on each call if there is no match? that could be expensive. can we just store that this class has null as admin to avoid that?